pkg/serverless/trace/trace.go

```diff
@@ -56,10 +56,8 @@ func (l *LoadConfig) Load() (*config.AgentConfig, error) {
 // Start starts the agent
 func (s *ServerlessTraceAgent) Start(enabled bool, loadConfig Load) {
 	if enabled {
-		// during hostname resolution the first step is to make a GRPC call which is timeboxed to a 2 seconds deadline
-		// in the serverless mode, we don't start the GRPC server so this call will fail and cause a 2 seconds delay
-		// by setting cmd_port to -1, this will cause the GRPC client to fail instantly
-		ddConfig.Datadog.Set("cmd_port", "-1")
+		// setting the hostname here saves about 1s in load time
+		ddConfig.Datadog.Set("hostname", "lambda")
```

Comment:
This test was failing because, despite skipping hostname resolution with gRPC, it was still taking > 1 second to determine hostname.

Solution is to avoid all hostname resolution by setting it before loading configuration.

Reply:
Note on this change.

We must be very careful that no host or hostname tags are set for any serverless invocations.

I confirmed this change does not set these tags by running a function with this new code and checking the UI. I also double checked the code to ensure that we're never calling `ddConfig.Datadog.GetString("hostname")`.